core/genesis_test.go

```diff
@@ -223,13 +223,16 @@ func TestReadWriteGenesisAlloc(t *testing.T) {
 			{1}: {Balance: big.NewInt(1), Storage: map[common.Hash]common.Hash{{1}: {1}}},
 			{2}: {Balance: big.NewInt(2), Storage: map[common.Hash]common.Hash{{2}: {2}}},
 		}
-		hash, _ = hashAlloc(alloc, false)
+		stateRoot, storageRootMessagePasser, _ = hashAlloc(alloc, false)
 	)
+	if storageRootMessagePasser.Cmp(common.Hash{}) != 0 {
```

Comment:
`common.Hash` is an array and the zero value can be compared in place like so, to avoid the `Cmp`:

```suggestion
	if storageRootMessagePasser != (common.Hash{}) {
```